include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -3697,6 +3697,9 @@ namespace internal
       const unsigned int                            face_no,
       const MatrixFreeFunctions::ShapeInfo<Number> &shape_info)
     {
+      if (dim == 1)
+        return;
```

Comment:
```suggestion
      if (dim == 1)
        {
          Assert(false, ExcInternalError());
          return;
        }
```
As you said, as this condition should never be met, I would suggest to clarify this be adding another assertion here right before the return statement. If you want to, you could add another in-source comment to why this return statement is necessary.